controller/handler/compliance-count.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * @module controller/handler/trial
+ */
+
+const database = require('../../model');
+const processPatient = require('../helper/process-patient');
+const processTrial = require('../helper/process-trial');
+const moment = require('moment');
+const getCount = require('../helper/process-compliance-count');
+
+/**
+ * A dashboard with an overview of a specific trial.
+ * @function getComplianceCount
+ * @param {row} currentTrial - Row
+ * @param {row} stages - Row
+ * @param {row} patients - Row
+ * @param {View} reply - Row
+ * @returns {View} Rendered page
+ */
+function getComplianceCount (currentTrial, stages, patients, reply) {
+    const startDate = moment().startOf('Week');
+
+    Promise
+        .all([
+            database.sequelize.query(
+                `
+        select patientId,
+        sum(case when state = 'expired' then 1 else 0 end) as expiredCount
```

Comment:
Well, this will only give the count of those patients whose survey had expired. But, this query fails to return how many surveys each patient has skipped over the past week which is required to categorize each patient into the red, green or yellow groups.